app/client/components/LinkingState.ts

```diff
@@ -54,105 +91,150 @@ function isSummaryOf(summary: TableRec, detail: TableRec): boolean {
  * @param srcColId: Name of the column that drives the target section, or null to use rowId.
  * @param tgtSection: RowModel for the section that's being driven.
  * @param tgtColId: Name of the reference column to auto-filter by, or null to auto-scroll.
- * @param byAllShown: For auto-filter, filter by all values in srcSection rather than only the
- *    value at the cursor. The user can use column filters on srcSection to control what's shown
- *    in the linked tgtSection.
+
  */
 export class LinkingState extends Disposable {
   // If linking affects target section's cursor, this will be a computed for the cursor rowId.
   public readonly cursorPos?: ko.Computed<UIRowId>;
 
-  // If linking affects filtering, this is a computed for the current filtering state, as a
-  // {[colId]: colValues} mapping, with a dependency on srcSection.activeRowId()
+  // If linking affects filtering, this is a computed for the current filtering state, including user-facing
+  // labels for filter values and types of the filtered columns
+  // with a dependency on srcSection.activeRowId()
+  // Is null if no filtering
+  public readonly filterState?: ko.Computed<FilterState>;
+
+  // filterColValues is a subset of the current filterState needed for filtering (subset of ClientQuery)
+  // {[colId]: colValues, [colId]: operations} mapping,
   public readonly filterColValues?: ko.Computed<FilterColValues>;
 
   // Get default values for a new record so that it continues to satisfy the current linking filters
   public readonly getDefaultColValues: () => any;
 
+  //Which case of linking we've got
+  public readonly linkTypeDescription: ko.Computed<LinkType>;
+
+  private _docModel: DocModel;
   private _srcSection: ViewSectionRec;
   private _srcTableModel: DataTableModel;
-  private _srcCol: ColumnRec;
   private _srcColId: string | undefined;
 
   constructor(docModel: DocModel, linkConfig: LinkConfig) {
     super();
     const {srcSection, srcCol, srcColId, tgtSection, tgtCol, tgtColId} = linkConfig;
+    this._docModel = docModel;
     this._srcSection = srcSection;
-    this._srcCol = srcCol;
     this._srcColId = srcColId;
     this._srcTableModel = docModel.dataTables[srcSection.table().tableId()];
     const srcTableData = this._srcTableModel.tableData;
 
-    if (tgtColId) {
-      const operation = isRefListType(tgtCol.type()) ? 'intersects' : 'in';
-      if (srcSection.parentKey() === 'custom') {
-        this.filterColValues = this._srcCustomFilter(tgtColId, operation);
-      } else if (srcColId) {
-        this.filterColValues = this._srcCellFilter(tgtColId, operation);
+    this.linkTypeDescription = ko.computed((): LinkType => {
```

Comment:
So my thinking with this was that the main linking logic could be expressed in terms of a big `switch(linkTypeDescription)`, which would mean all the tricky conditionals on colId and table type would be in one place, and the actual linking logic would be easier to write and read. I'm already using this in the changes to RightPanel.

I ended up not actually making the change because it would mean that linkTypeDescription would be critical to the logic of linking, but maybe it would be better to put it in now, especially with the redundancy of all the ifs around 20 lines down